### PR TITLE
Improve compatibility of wptmanifest with Python3

### PR DESCRIPTION
**Pull request by marmeladema**

### CODE REVIEW

File: tools/wptrunner/wptrunner/wptmanifest/backends/base.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import abc
+import six
```
Comment (jgraham) [inline tools/wptrunner/wptrunner/wptmanifest/backends/base.py]: Could we import the names we want directly i.e. `from six import iterkeys, iteritems, itervalues`?

**Review by jgraham — CHANGES_REQUESTED** [review]
In general this looks good, but we should ensure we import the symbols directly from six rather than going through a lookup on each use.

File: tools/wptrunner/wptrunner/wptmanifest/serializer.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import six
```
Comment (jgraham) [inline tools/wptrunner/wptrunner/wptmanifest/serializer.py]: Also import the relevant names here.

File: tools/wptrunner/wptrunner/wptmanifest/backends/base.py (hunk 1 above)
Comment (marmeladema) [reply tools/wptrunner/wptrunner/wptmanifest/backends/base.py]: Fixed!

File: tools/wptrunner/wptrunner/wptmanifest/serializer.py (hunk 2 above)
Comment (marmeladema) [reply tools/wptrunner/wptrunner/wptmanifest/serializer.py]: Fixed!

**Review by jgraham — APPROVED** [review]
Thanks!

File: tools/wptrunner/wptrunner/wptmanifest/serializer.py [hunk 3]
```diff
@@ -1,3 +1,5 @@
+from six import ensure_str, ensure_text
```
Comment (jgraham) [inline tools/wptrunner/wptrunner/wptmanifest/serializer.py]: Looks like we need to upgrade `six` to get this.

**Review by jgraham — CHANGES_REQUESTED** [review]

File: tools/wptrunner/wptrunner/wptmanifest/serializer.py (hunk 3 above)
Comment (marmeladema) [reply tools/wptrunner/wptrunner/wptmanifest/serializer.py]: Yes, i was looking at the same thing. What file do i have to edit?

Comment (marmeladema) [reply tools/wptrunner/wptrunner/wptmanifest/serializer.py]: It requires a `six` version `>=1.12.0` but I can see `six==1.13.0` in `tools/wptrunner/requirements.txt`.
Do i have to also specify it in all per browser requirements_<browser>.txt files?

Comment (jgraham) [reply tools/wptrunner/wptrunner/wptmanifest/serializer.py]: The issue is that our vendored copy is out of date. I'm investigating how to update it without breaking the world; it looks like something strange happened with the import.

Comment (jgraham) [reply tools/wptrunner/wptrunner/wptmanifest/serializer.py]: I just merged the PR to update six, I'll retrigger this once the other CI failures are fixed; then if it passes I think we're good.

Comment (marmeladema) [reply tools/wptrunner/wptrunner/wptmanifest/serializer.py]: Should I rebase?

Comment (jgraham) [reply tools/wptrunner/wptrunner/wptmanifest/serializer.py]: I think just close+reopen should work since we test the provisional merge commit into master.

**Review by jgraham — APPROVED** [review]

**Pull request merged**